src/main/java/ua/danit/final_project/services/storage/S3StorageService.java

```diff
@@ -0,0 +1,68 @@
+package ua.danit.final_project.services.storage;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import ua.danit.final_project.entities.Task;
+import ua.danit.final_project.entities.TaskImage;
+import ua.danit.final_project.repositories.TaskImageRepository;
+
+import javax.validation.constraints.NotNull;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.UUID;
+
+@Service
+@PropertySource("classpath:aws.properties")
+public class S3StorageService implements StorageService {
+
+  private final FileStorageProperties storageProperties;
+  private final TaskImageRepository taskImageRepository;
+
+  @Value("${aws.bucket}")
+  private String bucket;
+
+  @Autowired
+  public S3StorageService(FileStorageProperties storageProperties,
+                          TaskImageRepository taskImageRepository) {
+    this.storageProperties = storageProperties;
+    this.taskImageRepository = taskImageRepository;
+  }
+
+
+  @Override
+  public String storeFile(@NotNull MultipartFile file, Task task) throws IOException {
+
+    AmazonS3 s3 = storageProperties.getAmazonS3();
+
+    String key = "taskPhotos/" + UUID.randomUUID();
+    InputStream is = file.getInputStream();
+    //    s3.putObject(bucket, key, is, new ObjectMetadata());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=122&issues=AWYMwJKrvOWp7Yd6OrHr&open=AWYMwJKrvOWp7Yd6OrHr)

[ISSUE_KEY]: AWYMwJKrvOWp7Yd6OrHr